packages/flutter_tools/lib/src/base/net.dart

```diff
@@ -12,31 +13,46 @@ const int kNetworkProblemExitCode = 50;
 
 /// Download a file from the given URL and return the bytes.
 Future<List<int>> fetchUrl(Uri url) async {
-  printTrace('Downloading $url.');
+  int attempts = 0;
+  int duration = 1;
+  while (true) {
+    attempts += 1;
```

Comment:
I feel like we should have a `maxRetries` named param with a sensible default.

Reply:
Why? What should the user do if we exceed the max retries? Restart us?

Reply:
I guess I just like the aesthetics of a hard failure at some point, but I have no objective reason why